### PR TITLE
Update KV Credential Policy

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes Sync Flow of Key Vault Credential Policy

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/KeyVaultCredentialPolicy.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import com.azure.core.credential.TokenCredential;
 import com.azure.core.credential.TokenRequestContext;
 import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/KeyVaultCredentialPolicy.java:8]: This unused import is triggering a CheckStyle error.
```suggestion
```

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java [hunk 2]
```diff
@@ -235,7 +235,7 @@ public boolean authorizeRequestOnChallengeSync(HttpPipelineCallContext context,
         Optional<Object> contentLengthOptional = context.getData(KEY_VAULT_STASHED_CONTENT_LENGTH_KEY);
 
         if (request.getBody() == null && contentOptional.isPresent() && contentLengthOptional.isPresent()) {
-            request.setBody(BinaryData.fromObject(contentOptional.get()));
+            request.setBody((BinaryData) (contentOptional.get()));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java:238]: The CI pipeline tests have a few failures indicating we cannot cast things like `String` to `BinaryData`, is this expected?

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java [hunk 3]
```diff
@@ -218,7 +218,7 @@ public void authorizeRequestSync(HttpPipelineCallContext context) {
         // Do not overwrite previous contents if retrying after initial request failed (e.g. timeout).
         if (!context.getData(KEY_VAULT_STASHED_CONTENT_KEY).isPresent()) {
             if (request.getBody() != null) {
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java:220]: Should this be getBodyAsBinaryData too?

Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java]: yeah

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java (hunk 2 above)
Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java:239]: updated the tests

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java [hunk 4]
```diff
@@ -208,6 +208,7 @@ public void authorizeRequestSync(HttpPipelineCallContext context) {
                 .setTenantId(this.challenge.getTenantId());
 
             setAuthorizationHeaderSync(context, tokenRequestContext);
+            return;
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java:211]: Note: This is required, it was missing before when this policy got updated via #29730

File: sdk/keyvault/azure-security-keyvault-secrets/pom.xml [hunk 5]
```diff
@@ -47,6 +47,7 @@
     <!-- Configures the Java 9+ run to perform the required module exports, opens, and reads that are necessary for testing but shouldn't be part of the module-info. -->
     <javaModulesSurefireArgLine>
       --add-exports com.azure.core/com.azure.core.implementation.http=ALL-UNNAMED
+      --add-exports com.azure.core/com.azure.core.implementation.util=ALL-UNNAMED
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-secrets/pom.xml:50]: I take it you needed this for running local live tests?

**Review by vcolin7 — COMMENTED** [review]
We should add these changes to the KV Administration package as well.

File: sdk/keyvault/azure-security-keyvault-secrets/pom.xml (hunk 5 above)
Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-secrets/pom.xml:50]: for running unit tests in JDK 17 locally, yeah.
This config doesn't impact users, mainly for local tests running only.

**Comment by g2vinay** [comment]
> We should add these changes to the KV Administration package as well.

KV Administration package is autorest generated, so auto rest will take care of it when it adds sync stack to KV admin package.

File: sdk/keyvault/azure-security-keyvault-secrets/pom.xml (hunk 5 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-secrets/pom.xml:50]: Just as a note, we also needed this for Java 11, likely for anything using Java 9+? We're considering adding this to the parent POM. /cc @srnagar

**Comment by vcolin7** [comment]
> > We should add these changes to the KV Administration package as well.
> 
> KV Administration package is autorest generated, so auto rest will take care of it when it adds sync stack to KV admin package.

**Review by vcolin7 — APPROVED** [review]

**Comment by vcolin7** [comment]
Live and recorded tests passed. One live test failed due to not being able to fetch a dependency (a transient issue) and another one in a special environment takes too long to execute and times out due to unrelated reasons, so we're ignoring those.

**Pull request merged**